flink-runtime/src/main/java/org/apache/flink/runtime/io/network/NetworkEnvironment.java

```diff
@@ -331,6 +346,24 @@ private void registerInputMetrics(MetricGroup inputGroup, MetricGroup buffersGro
 		buffersGroup.gauge(METRIC_INPUT_POOL_USAGE, new InputBufferPoolUsageGauge(inputGates));
 	}
 
+	/**
+	 * Update consuming gate with newly available partition.
+	 *
+	 * @param partitionInfo telling where the partition can be retrieved from
+	 * @throws IOException IO problem by the update
+	 * @throws InterruptedException potentially blocking operation was interrupted
+	 * @throws PartitionException the input gate with the id from the partitionInfo is not found
+	 */
+	public void updatePartitionInfo(PartitionInfo partitionInfo)
+		throws IOException, InterruptedException, PartitionException {
+		IntermediateDataSetID intermediateResultPartitionID = partitionInfo.getIntermediateDataSetID();
+		SingleInputGate inputGate = inputGatesById.get(intermediateResultPartitionID);
+		if (inputGate == null) {
+			throw new PartitionException("No reader with ID " + intermediateResultPartitionID + " was found.");
```

Comment:
Should this maybe be an `IllegalStateException` because this should actually not happen?

Reply:
This is how it was reported before.. but I agree `IllegalStateException` fits better